docs/jetstream/operations.md

```diff
@@ -0,0 +1,129 @@
+
+Argo expects each step in the workflow to be a container. The existing Jetstream container, which has the Jetstream CLI installed, can be used for each of these steps.
+The full workflow definition is defined in [the `workflows/run.yaml` file](https://github.com/mozilla/jetstream/blob/main/jetstream/workflows/run.yaml).
+
+Depending on how Jetstream is invoked (`rerun`, `run-argo`, or `rerun_config_changed`), Jetstream will determine the dates and experiments that are to be analyzed and injects them as parameters into `run.yaml` before launching the workflow. Argo will create separate jobs for each experiment and each analysis date. Once the analysis is complete, data gets exported as JSON to GCS.
+
+### Parallelizing lower-level calculations
+
+In addition to running experiment analyses in parallel, [Dask](https://dask.org/) is used to parallelize lower-level calculations. The following steps could be executed in parallel:
+* Analyses for each analysis period (daily, 28day, weekly, overall)
+* Analyses for different segments
+* Calculating statistics defined for an experiment analysis
+
+The [`dask.delayed interface`](https://docs.dask.org/en/latest/delayed.html) is used to turn the functions executing these steps into tasks that are added to a task graph which executes these steps in parallel. Dask is configured to use as many cores as are available on the machine by default, with 1 worker for each core. [Multi-threading being avoided, instead processes are used](https://docs.dask.org/en/latest/scheduling.html#local-threads) since the code is dominated by Python code, otherwise there wouldn't be any speedup due Python's Global Interpreter Lock. To manually restrict the number of processes, the `JETSTREAM_PROCESSES` environment variable can be used.
+
+## Setup
+
+Jetstream is executed on the `jetstream` Kubernetes cluster in the `moz-fx-data-experiments` project which is set up following [Argo's installation guide](https://github.com/argoproj/argo/blob/master/docs/quick-start.md):
+* When creating or re-creating the cluster, Bigquery and Compute Engine read/write permissions need to be enabled
+* Installing Argo:
+	* Create an `argo` namespace: `kubectl create ns argo`
+	* Install commonly used components: `kubectl apply -n argo -f https://raw.githubusercontent.com/argoproj/argo/stable/manifests/quick-start-postgres.yaml`
+	* Create new `clusterrole`: `kubectl create rolebinding default-admin --clusterrole=admin --serviceaccount=argo:default --namespace=argo`
+* The [`jetstream` DAG in Airflow](https://github.com/mozilla/telemetry-airflow/blob/master/dags/jetstream.py) triggers the `run-argo` job daily and either requires Compute Engine API access or the parameters `cluster_ip` and `cluster_cert` need to be provided
+	* Currently the Airflow cluster does not have Compute Engine API access, so the cluster IP and certificate are stored as secrets and used for running Jetstream
+
+## Troubleshooting
```

Comment:
Do you think it would make sense to pull this out into a separate doc? I think this is all really good content but it'd be nice to be able to link the troubleshooting guide directly.

Reply:
Good point! I'll put this into a separate doc.